File: CustomViewObjc/ViewSource.m
//
//  ViewSource.m
//  CustomViewTest
//
//  Created by APP559 on 2014/7/28.
//  Copyright (c) 2014年 yume. All rights reserved.
//

#import "ViewSource.h"

@implementation ViewSource

+(ViewSource*) sharedInstance{
    static ViewSource *sharedInstance;
    static dispatch_once_t onceToken;
    
    dispatch_once(&onceToken, ^{
        sharedInstance = [ViewSource new];
    });
    
    return sharedInstance;
}

-(NSDictionary*)test1{
    return @{@"type":@"CustomWithXibLiveRenderKVC",
             @"labelLeft":@"l",
             @"labelRight":@"r",
             @"buttonLeft":@"ll",
             @"buttonRight":@"rr"};
}

-(NSDictionary*)test2{
    return @{@"type":@"CustomWithXibLiveRenderKVC",
             @"labelLeft":@"left",
             @"labelRight":@"right",
             @"buttonLeft":@"btn left",
             @"buttonRight":@"btn right"};
}

-(NSDictionary*)test3{
    return @{@"type":@"CustomWithXibLiveRenderKVC",
             @"labelLeft":@"左",
             @"labelRight":@"右",
             @"buttonLeft":@"按鈕左",
             @"buttonRight":@"按鈕右"};
}
@end

File: CustomViewSwift/ViewSource.swift
//
//  ViewSource.swift
//  apsmswift
//
//  Created by APP559 on 2014/8/11.
//  Copyright (c) 2014年 yume. All rights reserved.
//

import UIKit

var ViewSourceInstance: ViewSource {
get{
    return ViewSource.sharedInstance
}
}

class ViewSource: NSObject {
    
    class var sharedInstance: ViewSource {
    dispatch_once(&Inner.token) {
        Inner.instance = ViewSource()
        }
        return Inner.instance!
    }
    
    struct Inner {
        static var instance: ViewSource?
        static var token: dispatch_once_t = 0
    }
    
    func TestView1() -> NSDictionary{
        return ["type":"YumeButton",
            "buttonTitle":"Push Me!!"]
    }
    
    func TestView2() -> NSDictionary{
        return ["type":"YumeButton",
            "buttonTitle":"Push abc!!"]
    }
    
    func TestView3() -> NSDictionary{
        return ["type":"YumeButton",
            "buttonTitle":"Push def!!"]
    }
    
}
